### PR TITLE
Add delete site option to sites kebab menu

### LINKED ISSUE #95300: Feature Request: Add "Delete Site" to Kebab Menu at wordpress.com/sites
### What

Add "Delete Site" to Kebab Menu at wordpress.com/sites

### Why

Sometimes users end up with a big list of sites and need to delete/disconnect unwanted ones.  Right now they have to go to the site settings to delete, but this would help them speed up the process. And, now that people can restore deleted sites on their own, there's no much risk with putting it here.

### How

Image

### PR DESCRIPTION
**Pull request by madhusudhand**
Fixes: #95300 

## Proposed Changes

* Adds new option `Delete site` to the sites kebab menu
* Click takes to `/sites/settings/administration/${ site.slug }/delete-site`

## Testing Instructions

* Navigate to `/sites` and open the kebab menu of any of the site.
* It should have new option `Delete site`
* Click should take to site deletion page.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/sites/components/sites-dataviews/actions.tsx [hunk 1]
```diff
@@ -236,6 +236,10 @@ export const isActionEligible = (
 
 				return isNotAtomicJetpack( site ) || !! isDisconnectedJetpackAndNotAtomic( site );
 			};
+		case 'delete-site':
+			return ( site: SiteExcerptData ) => {
+				return ! site.is_deleted && ! isNotAtomicJetpack( site );
```
Comment (madhusudhand) [inline client/sites/components/sites-dataviews/actions.tsx]: Administration page has the following condition

https://github.com/Automattic/wp-calypso/blob/d916a343f24640fe512c227c43071a2751b80172/client/sites/settings/administration/tools/index.jsx#L238

Do we need to include the other checks such as isVip here?

Comment (okmttdhr) [reply client/sites/components/sites-dataviews/actions.tsx]: > isNotAtomicJetpack

This condition checks if the site is "not Atomic" but is a "Jetpack site." For the "Delete site" action, the condition should specifically verify whether the site is a Jetpack site, as that is the primary requirement.

> Do we need to include the other checks such as isVip here?

I'm not entirely familiar with the implications of the VIP check..., but I recommend keeping it for now to err on the side of caution unless we have clear confirmation that it's unnecessary. CC: @Automattic/lego 

Additionally, we should limit this action to admin users by including a `canManageOptions` check to ensure proper permissions. So the conditions would be like:

```suggestion
				return ! site.is_deleted && canManageOptions && ! isJetpack( site ) && ! isVip;
```

**Review by okmttdhr — COMMENTED** [review]
Thank you for working on this!

File: client/sites/components/sites-dataviews/actions.tsx (hunk 1 above)
Comment (madhusudhand) [reply client/sites/components/sites-dataviews/actions.tsx]: Updated the condition to `return ! site.is_deleted && canManageOptions && site.jetpack;` 👍 

@Automattic/lego Can you suggest on the `! isVip` requirement?

File: client/sites/components/sites-dataviews/actions.tsx [hunk 2]
```diff
@@ -558,6 +563,17 @@ export function useActions( {
 				},
 				isEligible: isActionEligible( 'migrate-to-wpcom', capabilities ),
 			},
+
+			{
+				id: 'delete-site',
+				label: __( 'Delete site' ),
+				callback: ( sites ) => {
+					const site = sites[ 0 ];
+					page( `/sites/settings/administration/${ site.slug }/delete-site` );
```
Comment (fushar) [inline client/sites/components/sites-dataviews/actions.tsx]: Let's use the existing URL (`/settings/delete-site/:slug`) for now, we have not released this URL yet.

File: client/sites/components/sites-dataviews/actions.tsx (hunk 1 above)
Comment (fushar) [reply client/sites/components/sites-dataviews/actions.tsx]: I think we must follow exactly the logic that shows the Delete Site option in Site Tools, i.e.:

> ( ! isJetpack || isAtomic ) && ! isVip && sitePurchasesLoaded

Currently, this PR does not show the Delete site option in /sites for Simple sites. However, if you go to one of the Simple site, then go to Settings -> General, you will see the Delete site option.

File: client/sites/components/sites-dataviews/actions.tsx (hunk 2 above)
Comment (madhusudhand) [reply client/sites/components/sites-dataviews/actions.tsx]: Updated.

File: client/sites/components/sites-dataviews/actions.tsx (hunk 1 above)
Comment (madhusudhand) [reply client/sites/components/sites-dataviews/actions.tsx]: I guess `sitePurchasesLoaded` is not needed as the grid is shown after the information is loaded. Please correct me if I am missing something.

**Review by okmttdhr — APPROVED** [review]
Confirmed the following ✅ 👍;

- "Delete site" works for Simple sites
- "Delete site" works for Atomic sites
- "Delete site" is hidden for Jetpack sites

File: client/sites/components/sites-dataviews/actions.tsx (hunk 1 above)
Comment (fushar) [reply client/sites/components/sites-dataviews/actions.tsx]: I think this is correct.

**Review by fushar — APPROVED** [review]

**Pull request merged**